build-system/common/ci.js

```diff
@@ -30,6 +93,24 @@ function isCiBuild() {
   return !!process.env.CI;
 }
 
+/**
+ * Returns true if this is a PR build.
+ * @return {boolean}
+ */
+function isPullRequestBuild() {
+  return serviceFunctionMap[getCiService()]['isPullRequestBuild']();
```

Comment:
Can `serviceFunctionMap` and `getCiService` be combined? Looks like those 2 functions and `ciService` are only called here. A quick suggestion - 

```
function isPullRequestBuild() {
  if ifXBuild() {
    return isXPullRequestBuild();
  }
  if ifYBuild() {
    return isYPullRequestBuild();
  }
  . . .
}
```

Reply:
The idea behind creating the map was to avoid the same set of `if` clauses in each function (several more will be added). I like the idea of combining `serviceFunctionMap` and `getCiService` though. Lemme know what you think of the new version.

Reply:
What do you mean about more `if` clauses being added?

Also hmm, when calling `isPullRequestBuild()` looks like `isTravisBuild()` is being called twice though, once in `ci.js` and once in `travis.js`. Thoughts on this?
```
function isPullRequestBuild() {
  return isXPullRequestBuild() || isYPullRequestBuild() || isZPullRequestBuild();
}
```

Reply:
> What do you mean about more `if` clauses being added?

In the end, there will be a dozen or so CI-agnostic functions similar to `isPullRequestBuild()`, e.g. `ciBuildUrl()`, `ciJobNumber()`, etc. I'm looking to avoid adding an `if X / if Y / if Z` block in each of them, hence the lookup table.

> Also hmm, when calling `isPullRequestBuild()` looks like `isTravisBuild()` is being called twice though, once in `ci.js` and once in `travis.js`. Thoughts on this?

This is another good catch. I've removed the extra check since it's no longer necessary.

> ```
> function isPullRequestBuild() {
>   return isXPullRequestBuild() || isYPullRequestBuild() || isZPullRequestBuild();
> }
> ```

I'm not too fond of this pattern because it won't work for non-boolean return types (e.g. SHA strings), and involves first calling the functions for service X and Y when we're on service Z, which the lookup table avoids.

Back to you for another round :)

Reply:
Makes sense! Thanks for humoring me.